metrics/registry.go

```diff
@@ -1,19 +1,50 @@
 package metrics
 
+var Reg *Registry
+
+const (
+	defaultMetricPort int = 9999
+	COUNTER           int = 0
+	GAUGE             int = 1
+	SUMMARY           int = 2
+	UNTYPED           int = 3
+	HISTOGRAM         int = 4
+)
+
+//generic metric structure
+type Metric struct {
+	MetricType  int
+	MetricName  string
+	MetricValue int64
+}
+
 type Registry struct {
 	provider RegistryProvider
 }
 
 // RegistryProvider interface that implements metric registry types
 type RegistryProvider interface {
 	init()
+	ingest(metric Metric)
 }
```

Comment:
we shall drop RegistryProvider as it's not serving much of a purpose, instead we can simplify Registry interface to start and ingest metrics. DI can be something we can focus when we need to support multiple monitoring sources.

in the suggestion, we will be using empty interface `interface {}` which is ideally not recommended as the failures will be thrown at runtime. in future, this can be replaced with generics.

```suggestion
type Registry interface {
        start(conf interface{})
	ingest(metric Metric)
}

var registry *PrometheusRegistry
```